packages/node_modules/@ciscospark/spark-core/src/interceptors/auth.js

```diff
@@ -78,9 +70,27 @@ export default class AuthInterceptor extends Interceptor {
         if (options.service) {
           return this.spark.internal.device.isService(options.service);
         }
-
         if (options.uri) {
-          return this.spark.internal.device.isServiceUrl(options.uri);
+          // If service Url then return or else check if whitelisted
+          return this.spark.internal.device.isServiceUrl(options.uri)
+            .then((isServiceUrl) => {
+              if (isServiceUrl) {
```

Comment:
@arun3528 please see my comment below. This part seems redundant. Shouldn't the function `isServiceUrl` be modified to accommodate this case? Otherwise the logic is split into two files.

Reply:
@mattnorris  currently the isServiceUrl only checks if the url exists in the serviceCatalog

The whitelisting check  against the whitelist domains was already here i just moved it around.

addAuthHeader will be false if the env variabled it set . There is one request for which we dont want to add the header at all . For that request we need to pass the header anyways 

We can have a call tomorrow . may be there is a better solution